server/src/test/java/org/apache/iotdb/db/integration/IoTDBCreateTimeseriesIT.java

```diff
@@ -0,0 +1,137 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Notice that, all test begins with "IoTDB" is integration test. All test which will start the
+ * IoTDB server should be defined as integration test.
+ */
+public class IoTDBCreateTimeseriesIT {
+  private Statement statement;
+
+  @Before
+  public void setUp() throws Exception {
+    EnvironmentUtils.envSetUp();
+
+    Class.forName(Config.JDBC_DRIVER_NAME);
+    Connection connection = DriverManager.
+        getConnection("jdbc:iotdb://127.0.0.1:6667/", "root", "root");
+    statement = connection.createStatement();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    EnvironmentUtils.cleanEnv();
+  }
+
+  /**
+   * Test creating a time series that is a prefix path of an existing time series
+   */
+  @Test
+  public void testCreateTimeseries1() throws Exception {
+    String timeSeries1 = "root.sg1.aa.bb";
+    String timeSeries2 = "root.sg1.aa.bb.cc";
+
+    statement.execute(
+        String.format("create timeseries %s with datatype=INT64, encoding=PLAIN, compression=SNAPPY", timeSeries1));
+    statement.execute(
+        String.format("create timeseries %s with datatype=INT64, encoding=PLAIN, compression=SNAPPY", timeSeries2));
+
+    EnvironmentUtils.stopDaemon();
+    setUp();
+
+    boolean hasResult = statement.execute("show timeseries");
+    Assert.assertTrue(hasResult);
+
+    List<String> resultList = new ArrayList<>();
+    try (ResultSet resultSet = statement.getResultSet()) {
+      while (resultSet.next()) {
+        String timeseries = resultSet.getString("timeseries");
+        resultList.add(timeseries);
+      }
+    }
+    Assert.assertEquals(2, resultList.size());
+
+    if (resultList.get(0).split("\\.").length < resultList.get(1).split("\\.").length) {
+      Assert.assertEquals(timeSeries1, resultList.get(0));
+      Assert.assertEquals(timeSeries2, resultList.get(1));
+    } else {
+      Assert.assertEquals(timeSeries2, resultList.get(0));
+      Assert.assertEquals(timeSeries1, resultList.get(1));
+    }
+
+  }
+
+  /**
+   * Test if creating a time series will cause the storage group with same name to disappear
+   */
+  @Test
+  public void testCreateTimeseries2() throws Exception {
+    String timeSeries = "root.sg1.a.b.c";
+
+    statement.execute(String.format("SET storage group TO %s", timeSeries));
+    try {
+      statement.execute(
+          String.format("create timeseries %s with datatype=INT64, encoding=PLAIN, compression=SNAPPY", timeSeries));
+    } catch (IoTDBSQLException ignored) {
+    }
+
+    EnvironmentUtils.stopDaemon();
+    setUp();
+
+    statement.execute("show timeseries");
+    Set<String> resultList = new HashSet<>();
+    try (ResultSet resultSet = statement.getResultSet()) {
+      while (resultSet.next()) {
+        String str = resultSet.getString("timeseries");
+        resultList.add(str);
+      }
+    }
+    Assert.assertFalse(resultList.contains(timeSeries));
```

Comment:
This is ok, another way is to assert the PathAlreadyExistException in the catch clause.